parser/parser.y

```diff
@@ -3335,9 +3336,33 @@ DefaultValueExpr:
 	NowSymOptionFraction
 |	SignedLiteral
 |	NextValueForSequence
+|	BuiltinFunction
```

Comment:
We should also update the comments above: https://dev.mysql.com/doc/refman/8.0/en/data-type-defaults.html.